src/VariationPriceHandler.php

```diff
@@ -0,0 +1,211 @@
+   *   the attributes used as value.
+   */
+  public function getAllUsedAttributes() {
+    $usedAttributes = [];
+    $types = [];
+    $variation_type_storage = $this->entityTypeManager->getStorage('commerce_product_variation_type');
+    $variation_types = $variation_type_storage->loadMultiple();
+    $attribute_field_map = $this->attributeFieldManager->getFieldMap();
+    foreach ($variation_types as $variation_type_id => $variation_type) {
+      $usedAttributes[$variation_type_id] = array_column($attribute_field_map[$variation_type_id], 'attribute_id');
+    }
+    return $usedAttributes;
+  }
+
+  /**
+   * Provides the total price of attribues attached to a variation.
+   *
+   * @param string $variation_type_id
+   *   The variation type machine name.
+   * @param Drupal\commerce_product\Entity\ProductVariation $variation
+   *   The commerce product variation.
+   *
+   * @return int
+   *   Sum of all attribute values associated with a variation.
+   */
+  public function getAttributePriceSum($variation_type_id, ProductVariation $variation) {
+    $attributeTotalPrice = 0;
+    $usedAttributes = $this->getAllUsedAttributes();
+    $variationTypeAttributes = $usedAttributes[$variation_type_id];
+    foreach ($variationTypeAttributes as $attribute) {
+      // The attribute field name of variation type will always be in the
+      // format of `attrubute_attribute_machine_name`.
+      $attributefieldName = 'attribute_' . $attribute;
+      $attributeValueID = $variation->get($attributefieldName)->getValue()['0']['target_id'];
+      $productAttributeValue = $this->entityManager->getStorage('commerce_product_attribute_value')->load($attributeValueID);
+      if (!empty($productAttributeValue->get('price')->getValue()['0']['number'])) {
+        $attributePrice = $productAttributeValue->get('price')->getValue()['0']['number'];
+        $attributeTotalPrice += $attributePrice;
+      }
+    }
+    return $attributeTotalPrice;
+  }
+
+  /**
+   * Updates the product variation price
+   *
+   * @param Drupal\commerce_product\Entity\ProductVariation $variation
+   *   The commerce product variation.
+   *
+   * @return Drupal\commerce_price\Price
+   *   The price object.
+   */
+  public function getProductVariationPrice(ProductVariation $variation) {
+    $product_id = $variation->get('product_id')->getValue()['0']['target_id'];
+    $product = $this->entityManager->getStorage('commerce_product')->load($product_id);
+    $productBaseCost = $product->get('field_price')->getValue()['0']['number'];
+
+    // Get the total price of all attributes associated with the product
+    // variation type.
+    $attributePrice = $this->getAttributePriceSum($variation->bundle(), $variation);
+    $newPrice = $productBaseCost + $attributePrice;
+    $currency = $product->get('field_price')->getValue()['0']['currency_code'];
+    $newPrice = new Price((string) $newPrice, $currency);
+    return $newPrice;
+  }
+
+}
```

Comment:
**CHANGE REQUEST**

Let's rename this function to `recalculatePrice`. Also, this is the 2nd API method (for external use) so move it to the top with the other one before protected methods.

Reply:
Renamed to `recalculateVariationPrice`